File: solutions/zig/difference-of-squares/1/difference_of_squares.zig
pub fn squareOfSum(number: usize) usize {
    const sum = sumOfN(number);
    return sum * sum;
}

pub fn sumOfSquares(number: usize) usize {
    return sumOfN(number) * ((2 * number) + 1) / 3;
}

pub fn differenceOfSquares(number: usize) usize {
    return squareOfSum(number) - sumOfSquares(number);
}

fn sumOfN(number: usize) usize {
    return (number * (number + 1)) / 2;
}

File: solutions/go/difference-of-squares/1/difference_of_squares.go
package differenceofsquares

func sumOfN(n int) int {
	return (n * (n + 1)) / 2
}

func SquareOfSum(n int) int {
	sum := sumOfN(n)
	return sum * sum
}

func SumOfSquares(n int) int {
	return sumOfN(n) * ((2 * n) + 1) / 3
}

func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}
